apps/designmanual-frontend/app/utils/sanity/usePreviewableData.ts

```diff
@@ -0,0 +1,42 @@
+import { useEffect, useState } from "react";
+import { useLoaderData } from "react-router";
+
+import { usePreviewSubscription } from "./usePreviewSubscription";
+import { filterDataToSingleItem } from "./utils";
+
+export type PreviewableLoaderData<Data extends { _id: string }> = {
+  /** The initial data from the server */
+  initialData: Data[];
+  /** Whether or not preview is enabled */
+  isPreview: boolean;
+  /** The GROQ query */
+  query: string | null;
+  /** Any query parameters */
+  queryParams: Record<string, unknown> | null;
+};
+/**
+ * Get automatically updating data whenever you're in preview mode.
+ *
+ * You need to pass in the type of Sanity data you want to use, and you'll get back an object with automatically updating data.
+ */
+export const usePreviewableData = <Data extends { _id: string }>() => {
```

Comment:
Om du ønsker å la denne ligge i /sanity/ mappen, så er det bedre å bruke parameteret til typen export type PreviewableLoaderData<Data extends { _id: string }>

Så forblir dette typesikkert. Vet ikke om du følger en guide, isåfall viser guiden feil praksis 😅

Reply:
Hooken er harvet fra tidligere dokumentasjon siden /docs og var sikker et forsøkt for å få preview i studio til å leve sammen med frontend url. Men vi bruker ikke preview i dette studio så adde ikke særlig funksjon og lagd bare støy. Jeg fjernet hooken og erstatte med data som er i loaders